### DISCUSSION TITLE
Was macOS 11 Big Sur support dropped in libomp?

### DISCUSSION THREAD

**Discussion by calebwherry** [Homebrew/discussions#4827] [Casks] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.12
ORIGIN: https://github.com/Homebrew/brew
HEAD: aa2a77b1f1ac1539e69e186bad1d569100448987
Last commit: 3 weeks ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e0ecef05329c3655654ad551d8cd45b13924e103
Core tap last commit: 13 days ago
Core tap branch: master
Core tap JSON: 22 Sep 00:15 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: ["--no-quarantine"]
HOMEBREW_COLOR: set
HOMEBREW_MAKE_JOBS: 3
HOMEBREW_NO_AUTO_UPDATE: set
HOMEBREW_NO_INSTALL_CLEANUP: set
Homebrew Ruby: 2.6.10 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.10_1/bin/ruby
CPU: 3-core 64-bit ivybridge
Clang: 13.0.0 build 1300
Git: 2.42.0 => /usr/local/bin/git
Curl: 8.1.2 => /usr/bin/curl
macOS: 11.7.10-x86_64
CLT: 13.2.0.0.1.1638488800
Xcode: 13.2.1 => /Applications/Xcode_13.2.1.app/Contents/Developer
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ruby@2.7

Warning: You have an unnecessary local Cask tap.
This can cause problems installing up-to-date casks.
Please remove it by running:
  brew untap homebrew/cask

Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/s/swiftlint.rb
```

### Description of issue

A PR went in yesterday that bumped libomp from 17.0.1 to 17.0.2. In this PR, macOS 11 Big Sur was removed. I cannot find release notes or anything relating to this removal? PR in question:

https://github.com/Homebrew/homebrew-core/pull/149284

My brew is running inside a macOS 11 GitHub runner.

Ideas? Was this announced and I missed it? Major OS removal in a patch release doesn't seem right...

### Relevant casks

libomp

**Comment by calebwherry** [comment]
Looks like I just missed the whole deprecation: answered here - https://github.com/Homebrew/homebrew-core/pull/149284#issuecomment-1747316304

**Comment by ZhongRuoyu** [comment, marked as answer]
This is not related to `libomp`. We (Homebrew) have dropped support for Big Sur when Sonoma was released. (We only support the latest three major macOS releases.) Since then, we no longer build bottles for Big Sur.

**Reply by manongjohn** [reply]
Slightly off topic but related....Why can't you support the versions Github Actions provides, which happens to be Big Sur?  A change like this has potential impact to Github Actions for those that build on the oldest supported GA environment for whatever reason.

I still build on Big Sur and now it tries to compile `qt@5` and my actions fail after 6 hours because it's taking so long to build. :(

**Reply by ZhongRuoyu** [reply]
@manongjohn:

Sorry to hear that, but GitHub Actions' remaining support for an outdated macOS version does not influence Homebrew's policy. (As a side note, we do not build our bottles with GitHub-hosted runners.) Rather, we don't support Big Sur simply because neither does Apple.

Also note that according to [GitHub Actions' image deprecation policy](https://github.com/actions/runner-images/tree/31a055838e0555e67b8987fb8719e3b37c9d644c?tab=readme-ov-file#image-deprecation-policy), the deprecation process of an OS's oldest available release (Big Sur) begins when a new release (Sonoma) is available. It seems they haven't started deprecating Big Sur yet, but this could happen anytime soon. And last year, [Catalina was deprecated](https://github.com/actions/runner-images/pull/5842) ~4 months *before* Apple's official release of Ventura.

**Reply by calebwherry** [reply]
@ZhongRuoyu 

Do you have a reference for the official policy of Apple deprecating older OSes? From what I can, Apple still supports Big Sur and is putting out security patches.

This might change soon but I haven’t been able to find anything about it being officially end of lifed by Apple.

**Reply by ZhongRuoyu** [reply]
As far as I know, Apple does not officially indicate their support policy for macOS; rather, this is deduced from [historical records](https://en.wikipedia.org/wiki/MacOS_version_history#Releases). Please also see https://github.com/Homebrew/brew/issues/12371.

**Reply by SMillerDev** [reply]
Actually Apple has recently stated that only the last version of macOS is fully supported. https://support.apple.com/guide/deployment/about-software-updates-depc4c80847a/web
